### PR TITLE
refactor(core): Decouple post workflow execute event from internal hooks (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
Redo of #10232

### CODE REVIEW

File: packages/cli/src/telemetry/telemetry-event-relay.service.ts [hunk 1]
```diff
@@ -118,6 +122,9 @@ export class TelemetryEventRelay {
 		this.eventService.on('workflow-saved', async (event) => {
 			await this.workflowSaved(event);
 		});
+		this.eventService.on('workflow-post-execute', async (event) => {
+			await this.workflowPostExecute(event);
+		});
```
Comment (netroy) [inline packages/cli/src/telemetry/telemetry-event-relay.service.ts:127]: considering how frequently this pattern is repeated in this file, and likely will be repeated in other services as we start using more EventEmitters, I wonder if it makes sense to create a utility function to pass a map of eventName to handlerName, and then set these all up in there instead.

File: packages/cli/src/executions/execution-recovery.service.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@ import { Push } from '@/push';
 import { jsonStringify, sleep } from 'n8n-workflow';
 import { ExecutionRepository } from '@db/repositories/execution.repository';
 import { getWorkflowHooksMain } from '@/WorkflowExecuteAdditionalData'; // @TODO: Dependency cycle
```
Comment (netroy) [inline packages/cli/src/executions/execution-recovery.service.ts:5]: is this still a Dependency cycle?

**Review by netroy — APPROVED** [review]
Looks better now. Thanks

File: packages/cli/src/executions/execution-recovery.service.ts (hunk 2 above)
Comment (ivov) [reply packages/cli/src/executions/execution-recovery.service.ts:5]: Unfortunately yes, but pretty small:
- `ExecutionRecoveryService` → Dependency cycle via `@/eventbus/MessageEventBus/MessageEventBus:73`
- `MessageEventBus` → Dependency cycle via `@/WorkflowExecuteAdditionalData:5`

**Pull request merged**